clientd3d/client.c

```diff
@@ -105,6 +105,18 @@ LRESULT CALLBACK WndProc (HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
 
 	switch (message)
 	{
+		case WM_MOUSEWHEEL:
+		{
+				if (state == STATE_GAME)
+				{
+						int zDelta = GET_WHEEL_DELTA_WPARAM(wParam);
+						int direction = (zDelta > 0) ? 1 : -1; // Scroll up is positive, scroll down is negative
```

Comment:
Do you want to make the amount of zoom proportional to how far the wheel has moved?

Reply:
**Thinking out loud:**

I think my solution might end up a bit more complex than what I have here now. My mouse (notch scrolling) emits fixed increments of 120. To speed up the zoom when the wheel is scrolled quickly, I suppose I'd need to record consecutive scroll events and adjust zoom speed accordingly.

Then there's smooth scrolling mouse wheel support and I'd need to track down one of these mice to test.